src/Files.App/ViewModels/ItemViewModel.cs

```diff
@@ -2084,7 +2100,45 @@ private async Task AddFileOrFolderAsync(ListedItem? item)
 			enumFolderSemaphore.Release();
 		}
 
-		private async Task<ListedItem?> AddFileOrFolderAsync(string fileOrFolderPath)
+		private async Task AddFileOrFolderAsync(ListedItem? item, int index = -1)
+		{
+			if (item is null)
+				return;
+
+			try
+			{
+				await enumFolderSemaphore.WaitAsync(semaphoreCTS.Token);
+			}
+			catch (OperationCanceledException)
+			{
+				return;
+			}
+
+			if (!filesAndFolders.Any(x => x.ItemPath.Equals(item.ItemPath, StringComparison.OrdinalIgnoreCase))) // Avoid adding duplicate items
+			{
+				if (index == -1)
+				{
+					filesAndFolders.Add(item);
+				}
+				else
+				{
+					filesAndFolders.Insert(index, item);
+				}
+				if (UserSettingsService.FoldersSettingsService.AreAlternateStreamsVisible)
```

Comment:
```suggestion
				if (index == -1)
					filesAndFolders.Add(item);
				else
					filesAndFolders.Insert(index, item);

				if (UserSettingsService.FoldersSettingsService.AreAlternateStreamsVisible)
```

You should avoid using `{}` for single line `ifs` (see <a href="https://files.community/docs/contributing/code-style">style guidelines</a>). Also, adding a new line to split the two ifs will make the code more readable